### PR TITLE
fix(tests): Import tests correctly for pulpTestAll method

### PR DESCRIPTION
**Pull request by JulianATA**
Hello!
Based on the PR #306 , we should fix the situation:
```python3
>>> import pulp
>>> pulp.pulpTestAll()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'pulp' has no attribute 'pulpTestAll'
```

The reason it failed is based on 2 reasons:
* pulp/tests/\_\_init\_\_.py didn't import the run_tests module
* pulp/\_\_init\_\_.py didn't import the tests package

The modules doesn't share the same name with their packages are [submodules](https://docs.python.org/3/reference/import.html#submodules).
We should import submodules in \_\_init\_\_.py.
* The run_tests module does not share the name `tests` with its package.

So, I got 2 solutions here:
1. Import submodules in pulp/tests/\_\_init\_\_.py:
    * Which is provided in this PR.
    * Consistent with the pulp/\_\_init\_\_.py
    * Easy and less modified.
2. Rename the run_tests.py to tests.py:
    * Still need to import tests in pulp/\_\_init\_\_.py

### CODE REVIEW

File: pulp/__init__.py [hunk 1]
```diff
@@ -36,6 +36,7 @@
 from .apis import *
 from .utilities import *
 from .constants import *
+from .tests import *
```
Comment (pchtsp) [inline pulp/__init__.py:39]: I think I prefer only exposing `pulpTestAll` just to be safe so:

`from .tests import pulpTestAll`

File: pulp/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from .run_tests import *
```
Comment (pchtsp) [inline pulp/tests/__init__.py:1]: while you're at it, add a new line here.

File: pulp/__init__.py (hunk 1 above)
Comment (JulianATA) [reply pulp/__init__.py]: You are right. It is better to expose less members of package.

File: pulp/tests/__init__.py (hunk 2 above)
Comment (JulianATA) [reply pulp/tests/__init__.py]: Same, modified to import only pulpTestAll.

**Comment by JulianATA** [comment]
The most easy to intuitive way to check:
```python
>>> import pulp
>>> pulp.pulpTestAll()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'pulp' has no attribute 'pulpTestAll'
```
It'll return error while calling `pulp.pulpTestAll()` before.

```python
>>> import pulp
>>> pulp.pulpTestAll()
Solver <class 'pulp.apis.coin_api.PULP_CBC_CMD'> available
Solver <class 'pulp.apis.cplex_api.CPLEX_DLL'> unavailable
Solver <class 'pulp.apis.cplex_api.CPLEX_CMD'> unavailable
.
.
.
Ran 130 tests in 13.370s

OK
```
It'll log the testing messages after.

* For creating an unit test for it, we could use the built-in [`hasattr()`](https://docs.python.org/3/library/functions.html#hasattr) function.

> hasattr(object, name)¶
>> The arguments are an object and a string. The result is True if the string is the name of one of the object’s attributes, False if not. (This is implemented by calling getattr(object, name) and seeing whether it raises an AttributeError or not.)

The unit test created by hasattr():
```python
import pulp
assert(hasattr(pulp,'pulpTestAll'))
```
It'll log out assertion error before:
```bash
$ python3 test.py
Traceback (most recent call last):
  File "test.py", line 2, in <module>
   assert(hasattr(pulp,'pulpTestAll'))
AssertionError
```
It'll pass without any message after.
```bash
$ python3 test.py
```
Thanks for your checking!

**Comment by pchtsp** [comment]
Thanks again. You're right: for some reason I though that in order to run that kind of test we would need to have `pulp` installed in the environment, but it's not necessary apparently.

I propose you add a new unit test in `tests/test_pulp.py` like so:

```python
def test_pulpTestAll(self):
    """
    Test the availability of the function pulpTestAll
    """
    from pulp import pulpTestAll
```

I can confirm it fails without the changes and it runs ok with the changes.

**Comment by JulianATA** [comment]
@pchtsp , this is a very clean way to confirm! The proposed test is added.
Thanks!

**Comment by JulianATA** [comment]
Conflict resolved!

**Pull request merged**